spinalcordtoolbox/vertebrae/detect_c2c3.py

```diff
@@ -78,8 +78,7 @@ def detect_c2c3(nii_im, nii_seg, contrast, nb_sag_avg=7.0, verbose=1):
     # Run detection
     logger.info('Run C2-C3 detector...')
     os.environ["FSLOUTPUTTYPE"] = "NIFTI_PAIR"
-    cmd_detection = 'isct_spine_detect -ctype=dpdt "%s" "%s" "%s"' % \
-                    (path_model, 'data_midSlice', 'data_midSlice_pred')
+    cmd_detection = ['isct_spine_detect', '-ctype=dpdt', path_model, 'data_midSlice', 'data_midSlice_pred']
     # The command below will fail, but we don't care because it will output an image (prediction), which we
     # will use later on.
     s, o = run_proc(cmd_detection, verbose=0, is_sct_binary=True, raise_exception=False)
```

Comment:
Here is a brief explanation of why the path got split in the first place:

- `subprocess` can accept arguments as either a string or a list.
- Normally, `subprocess` will only split spaces if `shell=True`. If `shell=False`, then I think that we would be able to leave the path as-is.
- However, `run_proc` contains the following line of code to determine whether `subprocess` should be called with `shell=True`:
   https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/52131da66f91ec2fd3d0f2d1fd707cebbdb7dfd6/spinalcordtoolbox/utils/sys.py#L331
   And, since a string was passed before, then `shell=True`.
- By turning the string into a list of args, now `shell=False`, which avoids the path-splitting behavior.
- Alternatively, we could choose to put double-quotes into the argument string, which would also solve the whitespace issue.

I like what you've done, though! Your changes make perfect sense to me. :)

Reply:
Thank you @joshuacwnewton for the explanation and for this pull approval.